pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -81,25 +83,84 @@ func (r *rbac) createResources(ctx context.Context) error {
 
 	logger := logging.FromContext(ctx)
 
-	// fetch the list of all namespaces which doesn't have label
-	// `openshift-pipelines.tekton.dev/namespace-ready: <release-version>`
-	namespaces, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s != %s", namespaceVersionLabel, r.version),
-	})
+	// list of namespaces rbac resources need to be created
+	var rbacNamespaces []corev1.Namespace
+	// list of namespaces rbac resources need not to be created
+	var nonRbacNamespaces []corev1.Namespace
+
+	// list all namespaces
+	allNamespaces, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	// list of namespaces rbac resources need to be created
-	var rbacNamespaces []corev1.Namespace
-
-	// filter namespaces:
-	// ignore ns with name passing regex `^(openshift|kube)-`
-	for _, n := range namespaces.Items {
+	for _, n := range allNamespaces.Items {
 		if ignore := nsRegex.MatchString(n.GetName()); ignore {
 			continue
 		}
-		rbacNamespaces = append(rbacNamespaces, n)
+		v, ok := n.GetLabels()[disableRbacInjection]
+		if r.disableRbacClusterWide {
+			if ok {
+				if v == "false" || v == "" {
+					rbacNamespaces = append(rbacNamespaces, n)
+					continue
+				} else {
+					nonRbacNamespaces = append(nonRbacNamespaces, n)
+					continue
+				}
+				// kubectl label ns test openshift-pipelines.tekton.dev/disable-rbac-injection-
+				// will remove label itself from namespace so priority will be taken for global value
+			} else {
+				nonRbacNamespaces = append(nonRbacNamespaces, n)
+				continue
+			}
+		} else {
+			if ok {
+				if v == "true" {
+					nonRbacNamespaces = append(nonRbacNamespaces, n)
+					continue
+				} else {
+					rbacNamespaces = append(rbacNamespaces, n)
+					continue
+				}
+			} else {
+				rbacNamespaces = append(rbacNamespaces, n)
+				continue
+			}
+		}
+	}
```

Comment:
instead of nesting the checks, if we can make it 2 different checks, then it might make this easy to maitain in the future

how about

```
if disableRbacInjection explicitly set:
    if true
        add to disable list
    else
        add to enable list
    continue # as the label has precedence we don't have to check the value of config spec if label is set
    
if  r.disableRbacClusterWide = true
    add namespace to disablelist
else
    add namespace to enable list
```

Reply:
Yup updated :+1: